Objective-C/CBLDatabase.mm

```diff
@@ -957,8 +1001,65 @@ - (BOOL) saveDocument: (CBLDocument*)document
     return YES;
 }
 
-#pragma mark - RESOLVING REPLICATED CONFLICTS:
+#pragma mark - LIVE-QUERY
+
+- (void) addActiveLiveQuery: (CBLLiveQuery*)liveQuery {
+    CBL_LOCK(self) {
+        [self mustBeOpenAndNotClosing];
+        
+        if (!_activeLiveQueries)
+            _activeLiveQueries = [NSMutableSet new];
+        
+        [_activeLiveQueries addObject: liveQuery];
+    }
+}
+
+- (void) removeActiveLiveQuery: (CBLLiveQuery*)liveQuery {
+    CBL_LOCK(self) {
+        [_activeLiveQueries removeObject: liveQuery];
+        
+        if (_activeLiveQueries.count == 0)
+            [_closeCondition broadcast];
+    }
+}
+
+- (uint64_t) activeLiveQueryCount {
+    CBL_LOCK(self) {
+        return _activeLiveQueries.count;
+    }
+}
 
+#pragma mark - REPLICATOR
+
+// NOTE: These methods are called from the replicator thread.
```

Comment:
Is this true?  In this code, they seem to be called from the start methods... which is what I would expect...

Reply:
Yea, it's not accurate. I will change this.